core/src/test/java/org/apache/accumulo/core/clientImpl/ScannerBaseTest.java

```diff
@@ -0,0 +1,117 @@
+ * specific language governing permissions and limitations
+ * under the License.
+ */
+package org.apache.accumulo.core.clientImpl;
+
+import static org.easymock.EasyMock.createMock;
+import static org.easymock.EasyMock.expect;
+import static org.easymock.EasyMock.replay;
+import static org.junit.Assert.assertEquals;
+
+import java.util.HashMap;
+import java.util.Iterator;
+import java.util.Map;
+import java.util.function.BiConsumer;
+
+import org.apache.accumulo.core.client.ScannerBase;
+import org.apache.accumulo.core.data.Key;
+import org.apache.accumulo.core.data.Value;
+import org.apache.hadoop.io.Text;
+import org.junit.Before;
+import org.junit.Test;
+
+public class ScannerBaseTest {
+
+  private ScannerBase s;
+  private Map<Key,Value> scannerMap;
+  private Iterator<Map.Entry<Key,Value>> it;
+  private Key key;
+  private Value val;
+  private BiConsumer<Key,Value> keyValueConsumer;
+  private forEachTester fet;
+
+  private static class forEachTester {
+
+    private Map<Key,Value> map;
+
+    forEachTester(Map<Key,Value> map) {
+      this.map = map;
+    }
+
+    public void forEach(BiConsumer<? super Key,? super Value> keyValueConsumer) {
+      for (Map.Entry<Key,Value> entry : this.map.entrySet()) {
+        keyValueConsumer.accept(entry.getKey(), entry.getValue());
+      }
+    }
+  }
+
+  @Before
+  public void setUp() throws Exception {
+    s = createMock(ScannerBase.class);
+  }
+
+  @Test
+  public void testScannerBaseForEach() throws Exception {
+    key = new Key(new Text("a"), new Text("cf1"), new Text("cq1"));
+    val = new Value(new Text("v1"));
+    scannerMap = new HashMap<>();
+
+    scannerMap.put(key, val);
+
+    fet = new forEachTester(scannerMap);
+
+    it = scannerMap.entrySet().iterator();
+
+    expect(s.iterator()).andReturn(it).times(3);
+    replay(s);
+
+    Map<Key,Value> map = new HashMap<>();
+
+    class MyBiConsumer implements BiConsumer<Key,Value> {
+      @Override
+      public void accept(Key key, Value value) {
+        map.put(key, value);
+      }
+    }
+
+    keyValueConsumer = new MyBiConsumer();
+
+    fet.forEach(keyValueConsumer);
+
+    // Test the Scanner values put into the map via keyValueConsumer
+
+    for (Map.Entry<Key,Value> entry : map.entrySet()) {
+      Map.Entry<Key,Value> expectedEntry = s.iterator().next();
+      Key expectedKey = expectedEntry.getKey();
+      Value expectedValue = expectedEntry.getValue();
+
+      String expectedCf = expectedKey.getColumnFamily().toString();
+      String actualCf = entry.getKey().getColumnFamily().toString();
+
+      String expectedCq = expectedKey.getColumnQualifier().toString();
+      String actualCq = entry.getKey().getColumnQualifier().toString();
+
+      String expectedVal = expectedValue.toString();
+      String actualVal = entry.getValue().toString();
+
+      assertEquals(expectedCf, actualCf);
+      assertEquals(expectedCq, actualCq);
+      assertEquals(expectedVal, actualVal);
```

Comment:
You don't need to compare CF and CQ, you can compare Key equality directly. That will shorten the test code a bit. You also don't need to call `toString()` on anything.

```suggestion
      Map.Entry<Key,Value> expectedEntry = s.iterator().next();
      assertEquals(expectedEntry.getKey(), entry.getKey());
      assertEquals(expectedEntry.getValue(), entry.getValue());
```